javalib/src/main/scala/java/net/URI.scala

```diff
@@ -943,7 +943,7 @@ final class URI private () extends Comparable[URI] with Serializable {
     }
     if (!path.endsWith("/") && seglist.length > 0 &&
         include(seglist.length - 1)) {
-      newpath.deleteCharAt(newpath.length() - 1)
+      newpath.deleteCharAt(newpath.size - 1)
```

Comment:
I think the real issue here is that `newpath` uses a Scala `StringBuilder` instead of a `java.lang.StringBuilder`. This is an opportunity to address the root issue.